share/completions/terraform.fish

```diff
@@ -0,0 +1,74 @@
+function __fish_terraform_needs_command
+    set cmd (commandline -opc)
+    if [ (count $cmd) -eq 1 ]
```

Comment:
The idiomatic fish way to do this is `if not set -q cmd[2]` to avoid the subcommand and test. Also, we prefer `test` over `[`. The latter is there strictly to make it easier for people transitioning from POSIX 1003.1 shells.